// @arcgis/map-components/dist/components/arcgis-elevation-profile/customElement.js
import { c as t } from "../../chunks/runtime.js";
import { usePropertyChange as i } from "@arcgis/components-controllers";
import e from "@arcgis/core/widgets/ElevationProfile.js";
import { LitElement as s, createEvent as o, noShadowRoot as h } from "@arcgis/lumina";
import { m as n } from "../../chunks/useWidget.js";
import { css as l } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const r = l`@layer{arcgis-elevation-profile{display:block}}`, a = n(e);
class d extends s {
  constructor() {
    super(...arguments), this.widget = a(this), this.viewModel = this.widget.viewModel, this.autoDestroyDisabled = !1, this.effectiveUnits = this.viewModel.effectiveUnits, this.hideChart = this.widget.visibleElements.chart, this.hideClearButton = this.widget.visibleElements.clearButton, this.hideLegend = this.widget.visibleElements.legend, this.hideSelectButton = this.widget.visibleElements.selectButton, this.hideSettingsButton = this.widget.visibleElements.settingsButton, this.hideSketchButton = this.widget.visibleElements.sketchButton, this.hideUniformChartScalingToggle = this.widget.visibleElements.uniformChartScalingToggle, this.hideUnitSelector = this.widget.visibleElements.unitSelector, this.highlightEnabled = this.viewModel.highlightEnabled, this.hoveredChartPosition = this.viewModel.hoveredChartPosition, this.icon = this.widget.icon, this.input = this.widget.input, this.label = this.widget.label, this.position = "bottom-left", this.profiles = this.widget.profiles, this.progress = this.viewModel.progress, this.state = this.viewModel.state, this.uniformChartScaling = this.viewModel.uniformChartScaling, this.unit = this.widget.unit, this.unitOptions = this.widget.unitOptions, this.arcgisPropertyChange = i()("input", "effectiveUnits", "progress", "state"), this.arcgisReady = o();
  }
  static {
    this.properties = { autoDestroyDisabled: 5, effectiveUnits: 0, hideChart: 5, hideClearButton: 5, hideLegend: 5, hideSelectButton: 5, hideSettingsButton: 5, hideSketchButton: 5, hideUniformChartScalingToggle: 5, hideUnitSelector: 5, highlightEnabled: 5, hoveredChartPosition: 9, icon: 1, input: 0, label: 1, position: 1, profiles: 0, progress: 8, referenceElement: 1, state: 3, uniformChartScaling: 5, unit: 1, unitOptions: 0 };
  }
  static {
    this.shadowRootOptions = h;
  }
  static {
    this.styles = r;
  }
  // #endregion
  // #region Public Methods
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
  async start() {
    this.widget.viewModel.start();
  }
  async stop() {
    this.widget.viewModel.stop();
  }
}
t("arcgis-elevation-profile", d);
export {
  d as ArcgisElevationProfile
};
