// @esri/calcite-components-react/dist/components.js
/// <reference types="@esri/calcite-components" preserve="true" />
import React from "react";
import { createComponent } from "@lit/react";
import { makeReactWrapperFactory, getReactWrapperOptions } from "@arcgis/lumina";
const createWrapper = /*@__PURE__*/ makeReactWrapperFactory(React, createComponent);
export const CalciteAccordion = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-accordion", {}));
export const CalciteAccordionItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-accordion-item", {}));
export const CalciteAction = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-action", {}));
export const CalciteActionBar = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-action-bar", {
    onCalciteActionBarToggle: "calciteActionBarToggle",
}));
export const CalciteActionGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-action-group", {}));
export const CalciteActionMenu = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-action-menu", {
    onCalciteActionMenuOpen: "calciteActionMenuOpen",
}));
export const CalciteActionPad = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-action-pad", {
    onCalciteActionPadToggle: "calciteActionPadToggle",
}));
export const CalciteAlert = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-alert", {
    onCalciteAlertBeforeClose: "calciteAlertBeforeClose",
    onCalciteAlertBeforeOpen: "calciteAlertBeforeOpen",
    onCalciteAlertClose: "calciteAlertClose",
    onCalciteAlertOpen: "calciteAlertOpen",
}));
export const CalciteAutocomplete = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-autocomplete", {
    onCalciteAutocompleteBeforeClose: "calciteAutocompleteBeforeClose",
    onCalciteAutocompleteBeforeOpen: "calciteAutocompleteBeforeOpen",
    onCalciteAutocompleteChange: "calciteAutocompleteChange",
    onCalciteAutocompleteClose: "calciteAutocompleteClose",
    onCalciteAutocompleteOpen: "calciteAutocompleteOpen",
    onCalciteAutocompleteTextChange: "calciteAutocompleteTextChange",
    onCalciteAutocompleteTextInput: "calciteAutocompleteTextInput",
}));
export const CalciteAutocompleteItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-autocomplete-item", {}));
export const CalciteAutocompleteItemGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-autocomplete-item-group", {}));
export const CalciteAvatar = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-avatar", {}));
export const CalciteBlock = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-block", {
    onCalciteBlockBeforeClose: "calciteBlockBeforeClose",
    onCalciteBlockBeforeOpen: "calciteBlockBeforeOpen",
    onCalciteBlockClose: "calciteBlockClose",
    onCalciteBlockOpen: "calciteBlockOpen",
    onCalciteBlockSortHandleBeforeClose: "calciteBlockSortHandleBeforeClose",
    onCalciteBlockSortHandleBeforeOpen: "calciteBlockSortHandleBeforeOpen",
    onCalciteBlockSortHandleClose: "calciteBlockSortHandleClose",
    onCalciteBlockSortHandleOpen: "calciteBlockSortHandleOpen",
    onCalciteBlockToggle: "calciteBlockToggle",
}));
export const CalciteBlockGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-block-group", {
    onCalciteBlockGroupDragEnd: "calciteBlockGroupDragEnd",
    onCalciteBlockGroupDragStart: "calciteBlockGroupDragStart",
    onCalciteBlockGroupOrderChange: "calciteBlockGroupOrderChange",
}));
export const CalciteBlockSection = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-block-section", {
    onCalciteBlockSectionToggle: "calciteBlockSectionToggle",
}));
export const CalciteButton = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-button", {}));
export const CalciteCard = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-card", {
    onCalciteCardSelect: "calciteCardSelect",
}));
export const CalciteCardGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-card-group", {
    onCalciteCardGroupSelect: "calciteCardGroupSelect",
}));
export const CalciteCarousel = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-carousel", {
    onCalciteCarouselChange: "calciteCarouselChange",
    onCalciteCarouselPause: "calciteCarouselPause",
    onCalciteCarouselPlay: "calciteCarouselPlay",
    onCalciteCarouselResume: "calciteCarouselResume",
    onCalciteCarouselStop: "calciteCarouselStop",
}));
export const CalciteCarouselItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-carousel-item", {}));
export const CalciteCheckbox = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-checkbox", {
    onCalciteCheckboxChange: "calciteCheckboxChange",
}));
export const CalciteChip = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-chip", {
    onCalciteChipClose: "calciteChipClose",
    onCalciteChipSelect: "calciteChipSelect",
}));
export const CalciteChipGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-chip-group", {
    onCalciteChipGroupSelect: "calciteChipGroupSelect",
}));
export const CalciteColorPicker = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-color-picker", {
    onCalciteColorPickerChange: "calciteColorPickerChange",
    onCalciteColorPickerInput: "calciteColorPickerInput",
}));
export const CalciteColorPickerHexInput = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-color-picker-hex-input", {
    onCalciteColorPickerHexInputChange: "calciteColorPickerHexInputChange",
}));
export const CalciteColorPickerSwatch = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-color-picker-swatch", {}));
export const CalciteCombobox = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-combobox", {
    onCalciteComboboxBeforeClose: "calciteComboboxBeforeClose",
    onCalciteComboboxBeforeOpen: "calciteComboboxBeforeOpen",
    onCalciteComboboxChange: "calciteComboboxChange",
    onCalciteComboboxChipClose: "calciteComboboxChipClose",
    onCalciteComboboxClose: "calciteComboboxClose",
    onCalciteComboboxFilterChange: "calciteComboboxFilterChange",
    onCalciteComboboxOpen: "calciteComboboxOpen",
}));
export const CalciteComboboxItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-combobox-item", {
    onCalciteComboboxItemChange: "calciteComboboxItemChange",
}));
export const CalciteComboboxItemGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-combobox-item-group", {}));
export const CalciteDatePicker = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-date-picker", {
    onCalciteDatePickerChange: "calciteDatePickerChange",
    onCalciteDatePickerRangeChange: "calciteDatePickerRangeChange",
}));
export const CalciteDatePickerDay = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-date-picker-day", {}));
export const CalciteDatePickerMonth = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-date-picker-month", {}));
export const CalciteDatePickerMonthHeader = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-date-picker-month-header", {}));
export const CalciteDialog = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-dialog", {
    onCalciteDialogBeforeClose: "calciteDialogBeforeClose",
    onCalciteDialogBeforeOpen: "calciteDialogBeforeOpen",
    onCalciteDialogClose: "calciteDialogClose",
    onCalciteDialogOpen: "calciteDialogOpen",
    onCalciteDialogScroll: "calciteDialogScroll",
}));
export const CalciteDropdown = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-dropdown", {
    onCalciteDropdownBeforeClose: "calciteDropdownBeforeClose",
    onCalciteDropdownBeforeOpen: "calciteDropdownBeforeOpen",
    onCalciteDropdownClose: "calciteDropdownClose",
    onCalciteDropdownOpen: "calciteDropdownOpen",
    onCalciteDropdownSelect: "calciteDropdownSelect",
}));
export const CalciteDropdownGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-dropdown-group", {}));
export const CalciteDropdownItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-dropdown-item", {
    onCalciteDropdownItemSelect: "calciteDropdownItemSelect",
}));
export const CalciteFab = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-fab", {}));
export const CalciteFilter = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-filter", {
    onCalciteFilterChange: "calciteFilterChange",
}));
export const CalciteFlow = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-flow", {}));
export const CalciteFlowItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-flow-item", {
    onCalciteFlowItemBack: "calciteFlowItemBack",
    onCalciteFlowItemClose: "calciteFlowItemClose",
    onCalciteFlowItemScroll: "calciteFlowItemScroll",
    onCalciteFlowItemToggle: "calciteFlowItemToggle",
}));
export const CalciteGraph = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-graph", {}));
export const CalciteHandle = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-handle", {
    onCalciteHandleChange: "calciteHandleChange",
    onCalciteHandleNudge: "calciteHandleNudge",
}));
export const CalciteIcon = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-icon", {}));
export const CalciteInlineEditable = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-inline-editable", {
    onCalciteInlineEditableEditCancel: "calciteInlineEditableEditCancel",
    onCalciteInlineEditableEditConfirm: "calciteInlineEditableEditConfirm",
}));
export const CalciteInput = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-input", {
    onCalciteInputChange: "calciteInputChange",
    onCalciteInputInput: "calciteInputInput",
}));
export const CalciteInputDatePicker = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-input-date-picker", {
    onCalciteInputDatePickerBeforeClose: "calciteInputDatePickerBeforeClose",
    onCalciteInputDatePickerBeforeOpen: "calciteInputDatePickerBeforeOpen",
    onCalciteInputDatePickerChange: "calciteInputDatePickerChange",
    onCalciteInputDatePickerClose: "calciteInputDatePickerClose",
    onCalciteInputDatePickerOpen: "calciteInputDatePickerOpen",
}));
export const CalciteInputMessage = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-input-message", {}));
export const CalciteInputNumber = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-input-number", {
    onCalciteInputNumberChange: "calciteInputNumberChange",
    onCalciteInputNumberInput: "calciteInputNumberInput",
}));
export const CalciteInputText = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-input-text", {
    onCalciteInputTextChange: "calciteInputTextChange",
    onCalciteInputTextInput: "calciteInputTextInput",
}));
export const CalciteInputTimePicker = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-input-time-picker", {
    onCalciteInputTimePickerBeforeClose: "calciteInputTimePickerBeforeClose",
    onCalciteInputTimePickerBeforeOpen: "calciteInputTimePickerBeforeOpen",
    onCalciteInputTimePickerChange: "calciteInputTimePickerChange",
    onCalciteInputTimePickerClose: "calciteInputTimePickerClose",
    onCalciteInputTimePickerOpen: "calciteInputTimePickerOpen",
}));
export const CalciteInputTimeZone = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-input-time-zone", {
    onCalciteInputTimeZoneBeforeClose: "calciteInputTimeZoneBeforeClose",
    onCalciteInputTimeZoneBeforeOpen: "calciteInputTimeZoneBeforeOpen",
    onCalciteInputTimeZoneChange: "calciteInputTimeZoneChange",
    onCalciteInputTimeZoneClose: "calciteInputTimeZoneClose",
    onCalciteInputTimeZoneOpen: "calciteInputTimeZoneOpen",
}));
export const CalciteLabel = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-label", {}));
export const CalciteLink = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-link", {}));
export const CalciteList = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-list", {
    onCalciteListChange: "calciteListChange",
    onCalciteListDragEnd: "calciteListDragEnd",
    onCalciteListDragStart: "calciteListDragStart",
    onCalciteListFilter: "calciteListFilter",
    onCalciteListOrderChange: "calciteListOrderChange",
}));
export const CalciteListItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-list-item", {
    onCalciteListItemClose: "calciteListItemClose",
    onCalciteListItemSelect: "calciteListItemSelect",
    onCalciteListItemSortHandleBeforeClose: "calciteListItemSortHandleBeforeClose",
    onCalciteListItemSortHandleBeforeOpen: "calciteListItemSortHandleBeforeOpen",
    onCalciteListItemSortHandleClose: "calciteListItemSortHandleClose",
    onCalciteListItemSortHandleOpen: "calciteListItemSortHandleOpen",
    onCalciteListItemToggle: "calciteListItemToggle",
}));
export const CalciteListItemGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-list-item-group", {}));
export const CalciteLoader = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-loader", {}));
export const CalciteMenu = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-menu", {}));
export const CalciteMenuItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-menu-item", {
    onCalciteMenuItemSelect: "calciteMenuItemSelect",
}));
export const CalciteMeter = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-meter", {}));
export const CalciteModal = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-modal", {
    onCalciteModalBeforeClose: "calciteModalBeforeClose",
    onCalciteModalBeforeOpen: "calciteModalBeforeOpen",
    onCalciteModalClose: "calciteModalClose",
    onCalciteModalOpen: "calciteModalOpen",
}));
export const CalciteNavigation = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-navigation", {
    onCalciteNavigationActionSelect: "calciteNavigationActionSelect",
}));
export const CalciteNavigationLogo = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-navigation-logo", {}));
export const CalciteNavigationUser = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-navigation-user", {}));
export const CalciteNotice = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-notice", {
    onCalciteNoticeBeforeClose: "calciteNoticeBeforeClose",
    onCalciteNoticeBeforeOpen: "calciteNoticeBeforeOpen",
    onCalciteNoticeClose: "calciteNoticeClose",
    onCalciteNoticeOpen: "calciteNoticeOpen",
}));
export const CalciteOption = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-option", {}));
export const CalciteOptionGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-option-group", {}));
export const CalcitePagination = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-pagination", {
    onCalcitePaginationChange: "calcitePaginationChange",
}));
export const CalcitePanel = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-panel", {
    onCalcitePanelClose: "calcitePanelClose",
    onCalcitePanelScroll: "calcitePanelScroll",
    onCalcitePanelToggle: "calcitePanelToggle",
}));
export const CalcitePopover = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-popover", {
    onCalcitePopoverBeforeClose: "calcitePopoverBeforeClose",
    onCalcitePopoverBeforeOpen: "calcitePopoverBeforeOpen",
    onCalcitePopoverClose: "calcitePopoverClose",
    onCalcitePopoverOpen: "calcitePopoverOpen",
}));
export const CalciteProgress = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-progress", {}));
export const CalciteRadioButton = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-radio-button", {
    onCalciteRadioButtonChange: "calciteRadioButtonChange",
}));
export const CalciteRadioButtonGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-radio-button-group", {
    onCalciteRadioButtonGroupChange: "calciteRadioButtonGroupChange",
}));
export const CalciteRating = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-rating", {
    onCalciteRatingChange: "calciteRatingChange",
}));
export const CalciteScrim = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-scrim", {}));
export const CalciteSegmentedControl = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-segmented-control", {
    onCalciteSegmentedControlChange: "calciteSegmentedControlChange",
}));
export const CalciteSegmentedControlItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-segmented-control-item", {}));
export const CalciteSelect = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-select", {
    onCalciteSelectChange: "calciteSelectChange",
}));
export const CalciteSheet = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-sheet", {
    onCalciteSheetBeforeClose: "calciteSheetBeforeClose",
    onCalciteSheetBeforeOpen: "calciteSheetBeforeOpen",
    onCalciteSheetClose: "calciteSheetClose",
    onCalciteSheetOpen: "calciteSheetOpen",
}));
export const CalciteShell = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-shell", {}));
export const CalciteShellCenterRow = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-shell-center-row", {}));
export const CalciteShellPanel = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-shell-panel", {}));
export const CalciteSlider = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-slider", {
    onCalciteSliderChange: "calciteSliderChange",
    onCalciteSliderInput: "calciteSliderInput",
}));
export const CalciteSortHandle = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-sort-handle", {
    onCalciteSortHandleBeforeClose: "calciteSortHandleBeforeClose",
    onCalciteSortHandleBeforeOpen: "calciteSortHandleBeforeOpen",
    onCalciteSortHandleClose: "calciteSortHandleClose",
    onCalciteSortHandleMove: "calciteSortHandleMove",
    onCalciteSortHandleOpen: "calciteSortHandleOpen",
    onCalciteSortHandleReorder: "calciteSortHandleReorder",
}));
export const CalciteSortableList = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-sortable-list", {
    onCalciteListOrderChange: "calciteListOrderChange",
}));
export const CalciteSplitButton = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-split-button", {
    onCalciteSplitButtonPrimaryClick: "calciteSplitButtonPrimaryClick",
    onCalciteSplitButtonSecondaryClick: "calciteSplitButtonSecondaryClick",
}));
export const CalciteStack = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-stack", {}));
export const CalciteStepper = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-stepper", {
    onCalciteStepperChange: "calciteStepperChange",
    onCalciteStepperItemChange: "calciteStepperItemChange",
}));
export const CalciteStepperItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-stepper-item", {
    onCalciteStepperItemSelect: "calciteStepperItemSelect",
}));
export const CalciteSwitch = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-switch", {
    onCalciteSwitchChange: "calciteSwitchChange",
}));
export const CalciteTab = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tab", {}));
export const CalciteTabNav = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tab-nav", {
    onCalciteTabChange: "calciteTabChange",
}));
export const CalciteTabTitle = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tab-title", {
    onCalciteTabsActivate: "calciteTabsActivate",
    onCalciteTabsClose: "calciteTabsClose",
}));
export const CalciteTable = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-table", {
    onCalciteTablePageChange: "calciteTablePageChange",
    onCalciteTableSelect: "calciteTableSelect",
}));
export const CalciteTableCell = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-table-cell", {}));
export const CalciteTableHeader = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-table-header", {}));
export const CalciteTableRow = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-table-row", {
    onCalciteTableRowSelect: "calciteTableRowSelect",
}));
export const CalciteTabs = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tabs", {}));
export const CalciteTextArea = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-text-area", {
    onCalciteTextAreaChange: "calciteTextAreaChange",
    onCalciteTextAreaInput: "calciteTextAreaInput",
}));
export const CalciteTile = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tile", {
    onCalciteTileSelect: "calciteTileSelect",
}));
export const CalciteTileGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tile-group", {
    onCalciteTileGroupSelect: "calciteTileGroupSelect",
}));
export const CalciteTileSelect = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tile-select", {
    onCalciteTileSelectChange: "calciteTileSelectChange",
}));
export const CalciteTileSelectGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tile-select-group", {}));
export const CalciteTimePicker = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-time-picker", {
    onCalciteTimePickerChange: "calciteTimePickerChange",
}));
export const CalciteTip = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tip", {
    onCalciteTipDismiss: "calciteTipDismiss",
}));
export const CalciteTipGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tip-group", {}));
export const CalciteTipManager = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tip-manager", {
    onCalciteTipManagerClose: "calciteTipManagerClose",
}));
export const CalciteTooltip = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tooltip", {
    onCalciteTooltipBeforeClose: "calciteTooltipBeforeClose",
    onCalciteTooltipBeforeOpen: "calciteTooltipBeforeOpen",
    onCalciteTooltipClose: "calciteTooltipClose",
    onCalciteTooltipOpen: "calciteTooltipOpen",
}));
export const CalciteTree = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tree", {
    onCalciteTreeSelect: "calciteTreeSelect",
}));
export const CalciteTreeItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tree-item", {}));
