// @arcgis/map-components/dist/components/arcgis-floor-filter/customElement.js
import { c as e } from "../../chunks/runtime.js";
import i from "@arcgis/core/widgets/FloorFilter.js";
import { LitElement as s, createEvent as o, noShadowRoot as l } from "@arcgis/lumina";
import { m as a } from "../../chunks/useWidget.js";
import { css as r } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const h = r`@layer{arcgis-floor-filter{display:block}}`, n = a(i);
class c extends s {
  constructor() {
    super(...arguments), this.widget = n(this), this.viewModel = this.widget.viewModel, this.autoDestroyDisabled = !1, this.facility = this.widget.facility, this.headingLevel = this.widget.headingLevel, this.icon = this.widget.icon, this.label = this.widget.label, this.level = this.widget.level, this.longNames = this.widget.longNames, this.position = "bottom-left", this.site = this.widget.site, this.arcgisReady = o();
  }
  static {
    this.properties = { autoDestroyDisabled: 5, facility: 3, headingLevel: 9, icon: 1, label: 1, level: 3, longNames: 5, position: 1, referenceElement: 1, site: 3 };
  }
  static {
    this.shadowRootOptions = l;
  }
  static {
    this.styles = h;
  }
  // #endregion
  // #region Public Methods
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
  async updateWebDocument(t) {
    this.widget?.updateWebDocument(t);
  }
}
e("arcgis-floor-filter", c);
export {
  c as ArcgisFloorFilter
};
