// @arcgis/map-components/dist/components/arcgis-histogram-range-slider/customElement.js
import { c as i } from "../../chunks/runtime.js";
import { usePropertyChange as e } from "@arcgis/components-controllers";
import s from "@arcgis/core/widgets/HistogramRangeSlider.js";
import { LitElement as a, createEvent as r, noShadowRoot as o } from "@arcgis/lumina";
import { m as n } from "../../chunks/useWidget.js";
import { css as d } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const h = d`@layer{arcgis-histogram-range-slider{display:block}}`, l = n(s);
class g extends a {
  constructor() {
    super(...arguments), this.widget = l(this), this.viewModel = this.widget.viewModel, this.autoDestroyDisabled = !1, this.average = this.widget.average, this.barCreatedFunction = this.widget.barCreatedFunction, this.bins = this.widget.bins, this.dataLineCreatedFunction = this.widget.dataLineCreatedFunction, this.dataLines = this.widget.dataLines, this.excludedBarColor = this.widget.excludedBarColor, this.icon = this.widget.icon, this.includedBarColor = this.widget.includedBarColor, this.label = this.widget.label, this.labelFormatFunction = this.widget.labelFormatFunction, this.max = this.widget.max, this.min = this.widget.min, this.position = "bottom-left", this.precision = this.widget.precision, this.rangeType = this.widget.rangeType, this.standardDeviation = this.widget.standardDeviation, this.standardDeviationCount = this.widget.standardDeviationCount, this.state = this.viewModel.state, this.values = this.widget.values, this.arcgisPropertyChange = e()("state"), this.arcgisReady = r();
  }
  static {
    this.properties = { autoDestroyDisabled: 5, average: 9, barCreatedFunction: 0, bins: 0, dataLineCreatedFunction: 0, dataLines: 0, excludedBarColor: 0, icon: 1, includedBarColor: 0, label: 1, labelFormatFunction: 0, max: 9, min: 9, position: 1, precision: 9, rangeType: 1, referenceElement: 1, standardDeviation: 9, standardDeviationCount: 9, state: 3, values: 0 };
  }
  static {
    this.shadowRootOptions = o;
  }
  static {
    this.styles = h;
  }
  // #endregion
  // #region Public Methods
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
  /**
   * Generates a SQL where clause based on a given field and the slider's
   * rangeType
   */
  async generateWhereClause(t) {
    return this.widget.generateWhereClause(t);
  }
}
i("arcgis-histogram-range-slider", g);
export {
  g as ArcgisHistogramRangeSlider
};
