// @arcgis/map-components/dist/components/arcgis-scale-range-slider/customElement.js
import { c as e } from "../../chunks/runtime.js";
import { usePropertyChange as i } from "@arcgis/components-controllers";
import t from "@arcgis/core/widgets/ScaleRangeSlider.js";
import { LitElement as s, createEvent as a, noShadowRoot as l } from "@arcgis/lumina";
import { m as o } from "../../chunks/useWidget.js";
import { css as n } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const r = n`@layer{arcgis-scale-range-slider{display:block}}`, c = o(t);
class h extends s {
  constructor() {
    super(...arguments), this.widget = c(this), this.viewModel = this.widget.viewModel, this.autoDestroyDisabled = !1, this.disabled = this.widget.disabled, this.hidePreview = this.widget.visibleElements.preview, this.hideScaleMenusMaxScaleMenu = this.widget.visibleElements.scaleMenus.maxScaleMenu, this.hideScaleMenusMinScaleMenu = this.widget.visibleElements.scaleMenus.minScaleMenu, this.icon = this.widget.icon, this.label = this.widget.label, this.layer = this.widget.layer, this.maxScale = this.widget.maxScale, this.maxScaleLimit = this.widget.maxScaleLimit, this.minScale = this.widget.minScale, this.minScaleLimit = this.widget.minScaleLimit, this.mode = this.widget.mode, this.position = "bottom-left", this.region = this.widget.region, this.state = this.viewModel.state, this.arcgisPropertyChange = i()("state", "minScale", "maxScale"), this.arcgisReady = a();
  }
  static {
    this.properties = { autoDestroyDisabled: 5, disabled: 7, hidePreview: 5, hideScaleMenusMaxScaleMenu: 5, hideScaleMenusMinScaleMenu: 5, icon: 1, label: 1, layer: 0, maxScale: 9, maxScaleLimit: 9, minScale: 9, minScaleLimit: 9, mode: 1, position: 1, referenceElement: 1, region: 0, state: 3 };
  }
  static {
    this.shadowRootOptions = l;
  }
  static {
    this.styles = r;
  }
  // #endregion
  // #region Public Methods
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
}
e("arcgis-scale-range-slider", h);
export {
  h as ArcgisScaleRangeSlider
};
