// @arcgis/map-components/dist/components/arcgis-version-management-version-item/customElement.js
import { c as v } from "../../chunks/runtime.js";
import { html as c } from "lit-html";
import { LitElement as b, createEvent as f, noShadowRoot as x, nothing as I } from "@arcgis/lumina";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
class E extends b {
  constructor() {
    super(...arguments), this.arcgisManageVersion = f();
  }
  static {
    this.properties = { versionItemElementProps: 0 };
  }
  static {
    this.shadowRootOptions = x;
  }
  // #endregion
  // #region Rendering
  render() {
    const { versionItemElementProps: { currentUser: d, currentVersionIdentifier: m, hasAdvEditingUte: g, isVersionAdministrator: p, isVersioningApiAvailable: h, serviceUrl: t, strings: { actions: n }, versionInfo: e } } = this, s = e.versionIdentifier.name, V = s.substring(s.lastIndexOf(".") + 1), a = s.substring(0, s.lastIndexOf(".")), r = e.versionIdentifier.guid, l = `actionSwitchToVersion_${r}`, o = a.toUpperCase() === d.toUpperCase() && h && g || p, i = m.guid === e.versionIdentifier.guid, $ = c`<calcite-action appearance=transparent .disabled=${i} id=${l ?? I} icon=code-branch text @click=${() => {
      this.arcgisManageVersion.emit({ actionType: "changeVersion", serviceUrl: t, versionInfo: e });
    }} slot=actions-start></calcite-action>`, u = i ? void 0 : c`<calcite-tooltip overlay-positioning=fixed .referenceElement=${l}><span>${n.switchToVersion}</span></calcite-tooltip>`;
    return c`<calcite-list-item .description=${a} .label=${V} .value=${r}>${$}${u}<calcite-action-menu overlay-positioning=fixed slot=actions-end label><calcite-action .disabled=${i} icon=code-branch @click=${() => {
      this.arcgisManageVersion.emit({ actionType: "changeVersion", serviceUrl: t, versionInfo: e });
    }} scale=s .text=${n.switchToVersion} text-enabled></calcite-action><calcite-action .disabled=${!o || i} icon=code-branch-edit @click=${() => {
      this.arcgisManageVersion.emit({ actionType: "editVersion", serviceUrl: t, versionInfo: e });
    }} scale=s .text=${n.editVersion} text-enabled></calcite-action><calcite-action .disabled=${!o || o && i} icon=trash @click=${() => {
      this.arcgisManageVersion.emit({ actionType: "deleteVersion", serviceUrl: t, versionInfo: e });
    }} scale=s .text=${n.deleteVersion} text-enabled></calcite-action></calcite-action-menu></calcite-list-item>`;
  }
}
v("arcgis-version-management-version-item", E);
export {
  E as ArcgisVersionManagementVersionItem
};
