// @arcgis/map-components/dist/components/arcgis-coordinate-conversion/customElement.js
import { c as i } from "../../chunks/runtime.js";
import { usePropertyChange as s } from "@arcgis/components-controllers";
import o from "@arcgis/core/widgets/CoordinateConversion.js";
import { LitElement as n, createEvent as r, noShadowRoot as a } from "@arcgis/lumina";
import { m as d } from "../../chunks/useWidget.js";
import { css as h } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const l = h`@layer{arcgis-coordinate-conversion{display:block}}`, c = d(o);
class g extends n {
  constructor() {
    super(...arguments), this.widget = c(this), this.viewModel = this.widget.viewModel, this.autoDestroyDisabled = !1, this.conversions = this.widget.conversions, this.currentLocation = this.widget.currentLocation, this.formats = this.widget.formats, this.headingLevel = this.widget.headingLevel, this.hideCaptureButton = this.widget.visibleElements.captureButton, this.hideEditButton = this.widget.visibleElements.editButton, this.hideExpandButton = this.widget.visibleElements.expandButton, this.hideSettingsButton = this.widget.visibleElements.settingsButton, this.icon = this.widget.icon, this.label = this.widget.label, this.locationSymbol = this.widget.locationSymbol, this.mode = this.widget.mode, this.multipleConversionsDisabled = this.widget.multipleConversions, this.orientation = this.widget.orientation, this.position = "bottom-left", this.state = this.viewModel.state, this.storageDisabled = this.widget.storageEnabled, this.storageType = this.widget.storageType, this.arcgisPropertyChange = s()("currentLocation", "state"), this.arcgisReady = r();
  }
  static {
    this.properties = { autoDestroyDisabled: 5, conversions: 0, currentLocation: 0, formats: 0, headingLevel: 9, hideCaptureButton: 5, hideEditButton: 5, hideExpandButton: 5, hideSettingsButton: 5, icon: 1, label: 1, locationSymbol: 0, mode: 1, multipleConversionsDisabled: 5, orientation: 1, position: 1, referenceElement: 1, state: 3, storageDisabled: 5, storageType: 1 };
  }
  static {
    this.shadowRootOptions = a;
  }
  static {
    this.styles = l;
  }
  // #endregion
  // #region Public Methods
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
  async reverseConvert(t, e) {
    return await this.widget?.reverseConvert(t, e);
  }
}
i("arcgis-coordinate-conversion", g);
export {
  g as ArcgisCoordinateConversion
};
