// @arcgis/map-components/dist/components/arcgis-basemap-layer-list/customElement.js
import { c as e } from "../../chunks/runtime.js";
import { usePropertyChange as t } from "@arcgis/components-controllers";
import { reEmitEvent as i } from "@arcgis/components-controllers/accessor";
import s from "@arcgis/core/widgets/BasemapLayerList.js";
import { LitElement as r, createEvent as a, noShadowRoot as l } from "@arcgis/lumina";
import { m as o } from "../../chunks/useWidget.js";
import { css as h } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const d = h`@layer{arcgis-basemap-layer-list{display:block}}`, n = o(s);
class c extends r {
  constructor() {
    super(...arguments), this.widget = n(this), this.viewModel = this.widget.viewModel, this.autoDestroyDisabled = !1, this.baseFilterPredicate = this.widget.baseFilterPredicate, this.baseFilterText = this.widget.baseFilterText, this.baseListItemCreatedFunction = this.widget.baseListItemCreatedFunction, this.basemapTitle = this.widget.basemapTitle, this.catalogOptions = this.widget.catalogOptions, this.collapsed = this.widget.collapsed, this.dragEnabled = this.widget.dragEnabled, this.editingTitle = this.widget.editingTitle, this.filterPlaceholder = this.widget.filterPlaceholder, this.headingLevel = this.widget.headingLevel, this.hideBaseLayers = this.widget.visibleElements.baseLayers, this.hideFlow = this.widget.visibleElements.flow, this.hideHeading = this.widget.visibleElements.heading, this.hideReferenceLayers = this.widget.visibleElements.referenceLayers, this.hideStatusIndicators = this.widget.visibleElements.statusIndicators, this.icon = this.widget.icon, this.label = this.widget.label, this.minFilterItems = this.widget.minFilterItems, this.position = "bottom-left", this.referenceFilterPredicate = this.widget.referenceFilterPredicate, this.referenceFilterText = this.widget.referenceFilterText, this.referenceListItemCreatedFunction = this.widget.referenceListItemCreatedFunction, this.selectedItems = this.widget.selectedItems, this.selectionMode = this.widget.selectionMode, this.showCloseButton = this.widget.visibleElements.closeButton, this.showCollapseButton = this.widget.visibleElements.collapseButton, this.showEditTitleButton = this.widget.visibleElements.editTitleButton, this.showErrors = this.widget.visibleElements.errors, this.showFilter = this.widget.visibleElements.filter, this.showTemporaryLayerIndicators = this.widget.visibleElements.temporaryLayerIndicators, this.state = this.viewModel.state, this.visibilityAppearance = this.widget.visibilityAppearance, this.arcgisPropertyChange = t()("state"), this.arcgisReady = a(), this.arcgisTriggerAction = i(() => this.widget, "trigger-action");
  }
  static {
    this.properties = { autoDestroyDisabled: 5, baseFilterPredicate: 0, baseFilterText: 1, baseItems: 32, baseListItemCreatedFunction: 0, basemapTitle: 1, catalogLayerList: 32, catalogOptions: 0, collapsed: 7, dragEnabled: 5, editingTitle: 5, filterPlaceholder: 1, headingLevel: 9, hideBaseLayers: 5, hideFlow: 5, hideHeading: 5, hideReferenceLayers: 5, hideStatusIndicators: 5, icon: 1, label: 1, minFilterItems: 9, position: 1, referenceElement: 1, referenceFilterPredicate: 0, referenceFilterText: 1, referenceItems: 32, referenceListItemCreatedFunction: 0, selectedItems: 0, selectionMode: 1, showCloseButton: 5, showCollapseButton: 5, showEditTitleButton: 5, showErrors: 5, showFilter: 5, showTemporaryLayerIndicators: 5, state: 3, visibilityAppearance: 1 };
  }
  static {
    this.shadowRootOptions = l;
  }
  static {
    this.styles = d;
  }
  /** A collection of ListItems representing the baseLayers. */
  get baseItems() {
    return this.widget.baseItems;
  }
  /** The CatalogLayerList that displays a catalog layer's dynamic group layer. */
  get catalogLayerList() {
    return this.widget.catalogLayerList;
  }
  /** A collection of ListItems representing the referenceLayers. */
  get referenceItems() {
    return this.widget.referenceItems;
  }
  // #endregion
  // #region Public Methods
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
}
e("arcgis-basemap-layer-list", c);
export {
  c as ArcgisBasemapLayerList
};
