// @arcgis/map-components/dist/components/arcgis-histogram/customElement.js
import { c as t } from "../../chunks/runtime.js";
import { usePropertyChange as i } from "@arcgis/components-controllers";
import e from "@arcgis/core/widgets/Histogram.js";
import { LitElement as s, createEvent as a, noShadowRoot as o } from "@arcgis/lumina";
import { m as r } from "../../chunks/useWidget.js";
import { css as n } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const h = n`@layer{arcgis-histogram{display:block}}`, d = r(e);
class m extends s {
  constructor() {
    super(...arguments), this.widget = d(this), this.viewModel = this.widget.viewModel, this.autoDestroyDisabled = !1, this.average = this.widget.average, this.barCreatedFunction = this.widget.barCreatedFunction, this.bins = this.widget.bins, this.dataLineCreatedFunction = this.widget.dataLineCreatedFunction, this.dataLines = this.widget.dataLines, this.icon = this.widget.icon, this.label = this.widget.label, this.labelFormatFunction = this.widget.labelFormatFunction, this.layout = this.widget.layout, this.max = this.widget.max, this.min = this.widget.min, this.position = "bottom-left", this.state = this.widget.state, this.arcgisPropertyChange = i()("state"), this.arcgisReady = a();
  }
  static {
    this.properties = { autoDestroyDisabled: 5, average: 9, barCreatedFunction: 0, bins: 0, dataLineCreatedFunction: 0, dataLines: 0, icon: 1, label: 1, labelFormatFunction: 0, layout: 1, max: 9, min: 9, position: 1, referenceElement: 1, state: 3 };
  }
  static {
    this.shadowRootOptions = o;
  }
  static {
    this.styles = h;
  }
  // #endregion
  // #region Public Methods
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
}
t("arcgis-histogram", m);
export {
  m as ArcgisHistogram
};
