// @arcgis/map-components/dist/components/arcgis-area-measurement-3d/customElement.js
import { c as t } from "../../chunks/runtime.js";
import { usePropertyChange as e } from "@arcgis/components-controllers";
import s from "@arcgis/core/widgets/AreaMeasurement3D.js";
import { LitElement as i, createEvent as a, noShadowRoot as o } from "@arcgis/lumina";
import { m as r } from "../../chunks/useWidget.js";
import { css as n } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const l = n`@layer{arcgis-area-measurement-3d{display:block}}`, h = r(s);
class d extends i {
  constructor() {
    super(...arguments), this.widget = h(this), this.viewModel = this.widget.viewModel, this.analysis = this.widget.analysis, this.autoDestroyDisabled = !1, this.icon = this.widget.icon, this.label = this.widget.label, this.position = "bottom-left", this.state = this.viewModel.state, this.unit = this.widget.unit, this.unitOptions = this.widget.unitOptions, this.arcgisPropertyChange = e()("analysis", "state"), this.arcgisReady = a();
  }
  static {
    this.properties = { analysis: 0, autoDestroyDisabled: 5, icon: 1, label: 1, position: 1, referenceElement: 1, state: 3, unit: 1, unitOptions: 0 };
  }
  static {
    this.shadowRootOptions = o;
  }
  static {
    this.styles = l;
  }
  // #endregion
  // #region Public Methods
  async clear() {
    this.widget?.viewModel.clear();
  }
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
  async start() {
    this.widget?.viewModel.start();
  }
}
t("arcgis-area-measurement-3d", d);
export {
  d as ArcgisAreaMeasurement3d
};
