// @arcgis/map-components/dist/components/arcgis-building-explorer/customElement.js
import { c as e } from "../../chunks/runtime.js";
import { usePropertyChange as i } from "@arcgis/components-controllers";
import s from "@arcgis/core/widgets/BuildingExplorer.js";
import { LitElement as t, createEvent as l, noShadowRoot as o } from "@arcgis/lumina";
import { m as h } from "../../chunks/useWidget.js";
import { css as r } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const a = r`@layer{arcgis-building-explorer{display:block}}`, d = h(s);
class n extends t {
  constructor() {
    super(...arguments), this.widget = d(this), this.viewModel = this.widget.viewModel, this.autoDestroyDisabled = !1, this.headingLevel = this.widget.headingLevel, this.hideDisciplines = this.widget.visibleElements.disciplines, this.hideLevels = this.widget.visibleElements.levels, this.hidePhases = this.widget.visibleElements.phases, this.icon = this.widget.icon, this.label = this.widget.label, this.layers = this.widget.layers, this.level = this.viewModel.level, this.phase = this.viewModel.phase, this.position = "bottom-left", this.state = this.viewModel.state, this.arcgisPropertyChange = i()("level", "phase", "state"), this.arcgisReady = l();
  }
  static {
    this.properties = { autoDestroyDisabled: 5, headingLevel: 9, hideDisciplines: 5, hideLevels: 5, hidePhases: 5, icon: 1, label: 1, layers: 0, level: 0, phase: 0, position: 1, referenceElement: 1, state: 3 };
  }
  static {
    this.shadowRootOptions = o;
  }
  static {
    this.styles = a;
  }
  // #endregion
  // #region Public Methods
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
}
e("arcgis-building-explorer", n);
export {
  n as ArcgisBuildingExplorer
};
