// @arcgis/map-components/dist/components/arcgis-feature-form/customElement.js
import { c as e } from "../../chunks/runtime.js";
import { usePropertyChange as i } from "@arcgis/components-controllers";
import { reEmitEvent as t } from "@arcgis/components-controllers/accessor";
import s from "@arcgis/core/widgets/FeatureForm.js";
import { LitElement as r, createEvent as a, noShadowRoot as o } from "@arcgis/lumina";
import { m as h } from "../../chunks/useWidget.js";
import { css as m } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const l = m`@layer{arcgis-feature-form{display:block}}`, g = h(s);
class n extends r {
  constructor() {
    super(...arguments), this.widget = g(this), this.viewModel = this.widget.viewModel, this.autoDestroyDisabled = !1, this.feature = this.widget.feature, this.formTemplate = this.widget.formTemplate, this.groupDisplay = this.widget.groupDisplay, this.headingLevel = this.widget.headingLevel, this.icon = this.widget.icon, this.label = this.widget.label, this.layer = this.widget.layer, this.position = "bottom-left", this.state = this.viewModel.state, this.timeZone = this.widget.timeZone, this.arcgisPropertyChange = i()("state"), this.arcgisReady = a(), this.arcgisSubmit = t(() => this.widget, "submit"), this.arcgisValueChange = t(() => this.widget, "value-change");
  }
  static {
    this.properties = { autoDestroyDisabled: 5, feature: 0, formTemplate: 0, groupDisplay: 1, headingLevel: 9, icon: 1, label: 1, layer: 0, position: 1, referenceElement: 1, state: 3, timeZone: 1 };
  }
  static {
    this.shadowRootOptions = o;
  }
  static {
    this.styles = l;
  }
  // #endregion
  // #region Public Methods
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
  async getValues() {
    return this.widget?.getValues();
  }
  async submit() {
    this.widget?.submit();
  }
}
e("arcgis-feature-form", n);
export {
  n as ArcgisFeatureForm
};
