// @arcgis/map-components/dist/components/arcgis-shadow-cast/customElement.js
import { c as t } from "../../chunks/runtime.js";
import { usePropertyChange as e } from "@arcgis/components-controllers";
import s from "@arcgis/core/widgets/ShadowCast.js";
import { LitElement as o, createEvent as a, noShadowRoot as h } from "@arcgis/lumina";
import { m as d } from "../../chunks/useWidget.js";
import { css as r } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const l = r`@layer{arcgis-shadow-cast{display:block}}`, n = d(s);
class c extends o {
  constructor() {
    super(...arguments), this.widget = n(this), this.viewModel = this.widget.viewModel, this.autoDestroyDisabled = !1, this.discreteOptions = this.viewModel.discreteOptions, this.durationOptions = this.viewModel.durationOptions, this.endTimeOfDay = this.viewModel.endTimeOfDay, this.headingLevel = this.widget.headingLevel, this.hideColorPicker = this.widget.visibleElements.colorPicker, this.hideDatePicker = this.widget.visibleElements.datePicker, this.hideTimeRangeSlider = this.widget.visibleElements.timeRangeSlider, this.hideTimezone = this.widget.visibleElements.timezone, this.hideTooltip = this.widget.visibleElements.tooltip, this.hideVisualizationOptions = this.widget.visibleElements.visualizationOptions, this.icon = this.widget.icon, this.label = this.widget.label, this.position = "bottom-left", this.startTimeOfDay = this.viewModel.startTimeOfDay, this.state = this.viewModel.state, this.thresholdOptions = this.viewModel.thresholdOptions, this.utcOffset = this.viewModel.utcOffset, this.visualizationType = this.viewModel.visualizationType, this.arcgisPropertyChange = e()("state"), this.arcgisReady = a();
  }
  static {
    this.properties = { autoDestroyDisabled: 5, discreteOptions: 0, durationOptions: 0, endTimeOfDay: 9, headingLevel: 9, hideColorPicker: 5, hideDatePicker: 5, hideTimeRangeSlider: 5, hideTimezone: 5, hideTooltip: 5, hideVisualizationOptions: 5, icon: 1, label: 1, position: 1, referenceElement: 1, startTimeOfDay: 9, state: 3, thresholdOptions: 0, utcOffset: 9, visualizationType: 1 };
  }
  static {
    this.shadowRootOptions = h;
  }
  static {
    this.styles = l;
  }
  // #endregion
  // #region Public Methods
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
  async getDuration(i) {
    return await this.widget?.viewModel.getDuration(i);
  }
  async start() {
    this.widget?.viewModel.start();
  }
  async stop() {
    this.widget?.viewModel.stop();
  }
}
t("arcgis-shadow-cast", c);
export {
  c as ArcgisShadowCast
};
