// @arcgis/map-components/dist/components/arcgis-table-list/customElement.js
import { c as t } from "../../chunks/runtime.js";
import { usePropertyChange as e } from "@arcgis/components-controllers";
import { reEmitEvent as i } from "@arcgis/components-controllers/accessor";
import s from "@arcgis/core/widgets/TableList.js";
import { LitElement as o, createEvent as l, noShadowRoot as r } from "@arcgis/lumina";
import { m as a } from "../../chunks/useWidget.js";
import { css as h } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const d = h`@layer{arcgis-table-list{display:block}}`, n = a(s);
class m extends o {
  constructor() {
    super(...arguments), this.widget = n(this), this.viewModel = this.widget.viewModel, this.autoDestroyDisabled = !1, this.collapsed = this.widget.collapsed, this.dragEnabled = this.widget.dragEnabled, this.filterPlaceholder = this.widget.filterPlaceholder, this.filterPredicate = this.widget.filterPredicate, this.filterText = this.widget.filterText, this.icon = this.widget.icon, this.label = this.widget.label, this.listItemCreatedFunction = this.widget.listItemCreatedFunction, this.minDragEnabledItems = this.widget.minDragEnabledItems, this.minFilterItems = this.widget.minFilterItems, this.position = "bottom-left", this.selectedItems = this.widget.selectedItems, this.selectionMode = this.widget.selectionMode, this.state = this.viewModel.state, this.showCloseButton = this.widget.visibleElements.closeButton, this.showCollapseButton = this.widget.visibleElements.collapseButton, this.showErrors = this.widget.visibleElements.errors, this.showFilter = this.widget.visibleElements.filter, this.hideFlow = this.widget.visibleElements.flow, this.showHeading = this.widget.visibleElements.heading, this.hideStatusIndicators = this.widget.visibleElements.statusIndicators, this.showTemporaryTableIndicators = this.widget.visibleElements.temporaryTableIndicators, this.arcgisPropertyChange = e()("state"), this.arcgisReady = l(), this.arcgisTriggerAction = i(() => this.widget, "trigger-action");
  }
  static {
    this.properties = { autoDestroyDisabled: 5, collapsed: 7, dragEnabled: 5, filterPlaceholder: 1, filterPredicate: 0, filterText: 1, icon: 1, label: 1, listItemCreatedFunction: 0, minDragEnabledItems: 9, minFilterItems: 9, position: 1, referenceElement: 1, selectedItems: 0, selectionMode: 1, state: 3, tableItems: 32, showCloseButton: 5, showCollapseButton: 5, showErrors: 5, showFilter: 5, hideFlow: 5, showHeading: 5, hideStatusIndicators: 5, showTemporaryTableIndicators: 5 };
  }
  static {
    this.shadowRootOptions = r;
  }
  static {
    this.styles = d;
  }
  /** The collection of table ListItems displayed within the component. */
  get tableItems() {
    return this.widget.tableItems;
  }
  // #endregion
  // #region Public Methods
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
}
t("arcgis-table-list", m);
export {
  m as ArcgisTableList
};
