// @arcgis/map-components/dist/components/arcgis-feature-templates/customElement.js
import { c as t } from "../../chunks/runtime.js";
import { usePropertyChange as e } from "@arcgis/components-controllers";
import { reEmitEvent as i } from "@arcgis/components-controllers/accessor";
import s from "@arcgis/core/widgets/FeatureTemplates.js";
import { LitElement as o, createEvent as r, noShadowRoot as l } from "@arcgis/lumina";
import { m as a } from "../../chunks/useWidget.js";
import { css as h } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const n = h`@layer{arcgis-feature-templates{display:block}}`, c = a(s);
class d extends o {
  constructor() {
    super(...arguments), this.widget = c(this), this.viewModel = this.widget.viewModel, this.autoDestroyDisabled = !1, this.enableListScroll = this.widget.enableListScroll, this.filterFunction = this.widget.filterFunction, this.filterText = this.widget.filterText, this.groupBy = this.widget.groupBy, this.headingLevel = this.widget.headingLevel, this.hideFilter = this.widget.visibleElements.filter, this.icon = this.widget.icon, this.label = this.widget.label, this.layers = this.widget.layers, this.position = "bottom-left", this.selectionMode = this.widget.selectionMode, this.state = this.viewModel.state, this.arcgisPropertyChange = e()("state"), this.arcgisReady = r(), this.arcgisSelect = i(() => this.widget, "select");
  }
  static {
    this.properties = { autoDestroyDisabled: 5, enableListScroll: 5, filterFunction: 0, filterText: 1, groupBy: 1, headingLevel: 9, hideFilter: 5, icon: 1, label: 1, layers: 0, position: 1, referenceElement: 1, selectionMode: 1, state: 3 };
  }
  static {
    this.shadowRootOptions = l;
  }
  static {
    this.styles = n;
  }
  // #endregion
  // #region Public Methods
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
}
t("arcgis-feature-templates", d);
export {
  d as ArcgisFeatureTemplates
};
