// @arcgis/map-components/dist/components/arcgis-value-picker/customElement.js
import { c as i } from "../../chunks/runtime.js";
import { usePropertyChange as e } from "@arcgis/components-controllers";
import { reEmitEvent as t } from "@arcgis/components-controllers/accessor";
import s from "@arcgis/core/widgets/ValuePicker.js";
import { LitElement as a, createEvent as o, noShadowRoot as n } from "@arcgis/lumina";
import { m as h } from "../../chunks/useWidget.js";
import { css as r } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const l = r`@layer{arcgis-value-picker{display:block}}`, c = h(s);
class u extends a {
  constructor() {
    super(...arguments), this.widget = c(this), this.autoDestroyDisabled = !1, this.canNext = this.widget.canNext, this.canPlay = this.widget.canPlay, this.canPrevious = this.widget.canPrevious, this.caption = this.widget.caption, this.component = this.widget.component, this.hideNextButton = this.widget.visibleElements.nextButton, this.hidePlayButton = this.widget.visibleElements.playButton, this.hidePreviousButton = this.widget.visibleElements.previousButton, this.icon = this.widget.icon, this.label = this.widget.label, this.layout = this.widget.layout, this.loop = this.widget.loop, this.playRate = this.widget.playRate, this.position = "bottom-left", this.values = this.widget.values, this.arcgisAnimate = t(() => this.widget, "animate"), this.arcgisNext = t(() => this.widget, "next"), this.arcgisPause = t(() => this.widget, "pause"), this.arcgisPlay = t(() => this.widget, "play"), this.arcgisPrevious = t(() => this.widget, "previous"), this.arcgisPropertyChange = e()("values"), this.arcgisReady = o();
  }
  static {
    this.properties = { autoDestroyDisabled: 5, canNext: 7, canPlay: 7, canPrevious: 7, caption: 1, component: 0, hideNextButton: 5, hidePlayButton: 5, hidePreviousButton: 5, icon: 1, label: 1, layout: 1, loop: 5, playRate: 9, position: 1, referenceElement: 1, values: 0 };
  }
  static {
    this.shadowRootOptions = n;
  }
  static {
    this.styles = l;
  }
  // #endregion
  // #region Public Methods
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
  async next() {
    this.widget?.next();
  }
  async pause() {
    this.widget?.pause();
  }
  async play() {
    this.widget?.play();
  }
  async previous() {
    this.widget?.previous();
  }
}
i("arcgis-value-picker", u);
export {
  u as ArcgisValuePicker
};
