// @arcgis/map-components/dist/components/arcgis-attachments/customElement.js
import { c as t } from "../../chunks/runtime.js";
import { usePropertyChange as e } from "@arcgis/components-controllers";
import i from "@arcgis/core/widgets/Attachments.js";
import { LitElement as s, createEvent as a, noShadowRoot as d } from "@arcgis/lumina";
import { m as o } from "../../chunks/useWidget.js";
import { css as h } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const n = h`@layer{arcgis-attachments{display:block}}`, r = o(i);
class l extends s {
  constructor() {
    super(...arguments), this.widget = r(this), this.viewModel = this.widget.viewModel, this.autoDestroyDisabled = !1, this.capabilities = this.widget.capabilities, this.displayType = this.widget.displayType, this.graphic = this.widget.graphic, this.hideAddButton = this.widget.visibleElements.addButton, this.hideAddSubmitButton = this.widget.visibleElements.addSubmitButton, this.hideCancelAddButton = this.widget.visibleElements.cancelAddButton, this.hideCancelUpdateButton = this.widget.visibleElements.cancelUpdateButton, this.hideDeleteButton = this.widget.visibleElements.deleteButton, this.hideErrorMessage = this.widget.visibleElements.errorMessage, this.hideProgressBar = this.widget.visibleElements.progressBar, this.hideUpdateButton = this.widget.visibleElements.updateButton, this.icon = this.widget.icon, this.label = this.widget.label, this.position = "bottom-left", this.state = this.viewModel.state, this.submitting = this.widget.submitting, this.arcgisPropertyChange = e()("submitting", "state"), this.arcgisReady = a();
  }
  static {
    this.properties = { autoDestroyDisabled: 5, capabilities: 0, displayType: 1, graphic: 0, hideAddButton: 5, hideAddSubmitButton: 5, hideCancelAddButton: 5, hideCancelUpdateButton: 5, hideDeleteButton: 5, hideErrorMessage: 5, hideProgressBar: 5, hideUpdateButton: 5, icon: 1, label: 1, position: 1, referenceElement: 1, state: 3, submitting: 4 };
  }
  static {
    this.shadowRootOptions = d;
  }
  static {
    this.styles = n;
  }
  // #endregion
  // #region Public Methods
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
}
t("arcgis-attachments", l);
export {
  l as ArcgisAttachments
};
