// @arcgis/map-components/dist/components/arcgis-layer-list/customElement.js
import { c as i } from "../../chunks/runtime.js";
import { usePropertyChange as s } from "@arcgis/components-controllers";
import { reEmitEvent as a } from "@arcgis/components-controllers/accessor";
import o from "@arcgis/core/widgets/LayerList.js";
import { LitElement as r, createEvent as l, noShadowRoot as h } from "@arcgis/lumina";
import { m as d } from "../../chunks/useWidget.js";
import { css as n } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const g = n`@layer{arcgis-layer-list{display:block}}`, c = d(o);
class m extends r {
  constructor() {
    super(...arguments), this.widget = c(this), this.viewModel = this.widget.viewModel, this.autoDestroyDisabled = !1, this.catalogOptions = this.widget.catalogOptions, this.collapsed = this.widget.collapsed, this.dragEnabled = this.widget.dragEnabled, this.filterPlaceholder = this.widget.filterPlaceholder, this.filterPredicate = this.widget.filterPredicate, this.filterText = this.widget.filterText, this.headingLevel = this.widget.headingLevel, this.hideCatalogLayerList = this.widget.visibleElements.catalogLayerList, this.hideFlow = this.widget.visibleElements.flow, this.hideStatusIndicators = this.widget.visibleElements.statusIndicators, this.icon = this.widget.icon, this.knowledgeGraphOptions = this.widget.knowledgeGraphOptions, this.label = this.widget.label, this.listItemCreatedFunction = this.widget.listItemCreatedFunction, this.minDragEnabledItems = this.widget.minDragEnabledItems, this.minFilterItems = this.widget.minFilterItems, this.openedLayers = this.widget.openedLayers, this.position = "bottom-left", this.selectedItems = this.widget.selectedItems, this.selectionMode = this.widget.selectionMode, this.showCloseButton = this.widget.visibleElements.closeButton, this.showCollapseButton = this.widget.visibleElements.collapseButton, this.showErrors = this.widget.visibleElements.errors, this.showFilter = this.widget.visibleElements.filter, this.showHeading = this.widget.visibleElements.heading, this.showTemporaryLayerIndicators = this.widget.visibleElements.temporaryLayerIndicators, this.state = this.viewModel.state, this.visibilityAppearance = this.widget.visibilityAppearance, this.arcgisPropertyChange = s()("state"), this.arcgisReady = l(), this.arcgisTriggerAction = a(() => this.widget, "trigger-action");
  }
  static {
    this.properties = { autoDestroyDisabled: 5, catalogLayerList: 32, catalogOptions: 0, collapsed: 7, dragEnabled: 5, filterPlaceholder: 1, filterPredicate: 0, filterText: 1, headingLevel: 9, hideCatalogLayerList: 5, hideFlow: 5, hideStatusIndicators: 5, icon: 1, knowledgeGraphOptions: 0, label: 1, listItemCreatedFunction: 0, minDragEnabledItems: 9, minFilterItems: 9, openedLayers: 0, operationalItems: 32, position: 1, referenceElement: 1, selectedItems: 0, selectionMode: 1, showCloseButton: 5, showCollapseButton: 5, showErrors: 5, showFilter: 5, showHeading: 5, showTemporaryLayerIndicators: 5, state: 3, tableList: 32, visibilityAppearance: 1 };
  }
  static {
    this.shadowRootOptions = h;
  }
  static {
    this.styles = g;
  }
  /** The CatalogLayerList that displays a catalog layer's dynamic group layer. */
  get catalogLayerList() {
    return this.widget.catalogLayerList;
  }
  /** A collection of ListItems representing operational layers. */
  get operationalItems() {
    return this.widget.operationalItems;
  }
  /** The TableList that displays the tables associated with a KnowledgeGraphLayer. */
  get tableList() {
    return this.widget.tableList;
  }
  // #endregion
  // #region Public Methods
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
  async triggerAction(t, e) {
    this.widget?.triggerAction(t, e);
  }
}
i("arcgis-layer-list", m);
export {
  m as ArcgisLayerList
};
