// @arcgis/map-components/dist/components/arcgis-print/customElement.js
import { c as i } from "../../chunks/runtime.js";
import { usePropertyChange as s } from "@arcgis/components-controllers";
import { reEmitEvent as e } from "@arcgis/components-controllers/accessor";
import r from "@arcgis/core/widgets/Print.js";
import { LitElement as o, createEvent as a, noShadowRoot as l } from "@arcgis/lumina";
import { m as n } from "../../chunks/useWidget.js";
import { css as h } from "@lit/reactive-element/css-tag.js";
/*! All material copyright Esri, All Rights Reserved, unless otherwise specified.
See https://js.arcgis.com/4.32/esri/copyright.txt for details.
v4.32.14 */
const d = h`@layer{arcgis-print{display:block}}`, m = n(r);
class p extends o {
  constructor() {
    super(...arguments), this.widget = m(this), this.viewModel = this.widget.viewModel, this.allowedFormats = this.widget.allowedFormats, this.allowedLayouts = this.widget.allowedLayouts, this.autoDestroyDisabled = !1, this.exportedLinks = this.widget.exportedLinks, this.extraParameters = this.widget.extraParameters, this.headingLevel = this.widget.headingLevel, this.icon = this.widget.icon, this.includeDefaultTemplatesDisabled = this.widget.includeDefaultTemplates, this.label = this.widget.label, this.portal = this.widget.portal, this.position = "bottom-left", this.printServiceUrl = this.widget.printServiceUrl, this.showPrintAreaEnabled = this.widget.showPrintAreaEnabled, this.state = this.viewModel.state, this.templateCustomTextElements = this.widget.templateCustomTextElements, this.templateOptions = this.widget.templateOptions, this.arcgisComplete = e(() => this.widget, "complete"), this.arcgisPropertyChange = s()("state"), this.arcgisReady = a(), this.arcgisSubmit = e(() => this.widget, "submit");
  }
  static {
    this.properties = { allowedFormats: [1, { converter: (t) => typeof t == "string" && t !== "all" ? t.split(",") : t }], allowedLayouts: [1, { converter: (t) => typeof t == "string" && t !== "all" ? t.split(",") : t }], autoDestroyDisabled: 5, exportedLinks: 0, extraParameters: 1, headingLevel: 9, icon: 1, includeDefaultTemplatesDisabled: 5, label: 1, portal: 0, position: 1, printServiceUrl: 1, referenceElement: 1, showPrintAreaEnabled: 7, state: 3, templateCustomTextElements: 0, templateOptions: 0 };
  }
  static {
    this.shadowRootOptions = l;
  }
  static {
    this.styles = d;
  }
  // #endregion
  // #region Public Methods
  /** Permanently destroy the component */
  async destroy() {
    await this.manager.destroy();
  }
}
i("arcgis-print", p);
export {
  p as ArcgisPrint
};
